src/ast/binary_encoder.ml

```diff
@@ -762,14 +762,15 @@ let encode
 let write_file outfile filename content =
   let _dir, filename = Fpath.split_base filename in
   let filename = Fpath.set_ext "wasm" filename in
-  let filename = match outfile with Some name -> Fpath.v name | None -> filename in
-  let filename = Fpath.to_string filename in
-  let oc = Out_channel.open_bin filename in
-  Out_channel.output_string oc content;
-  Out_channel.close oc
+  match outfile with
+  | Some name -> Bos.OS.File.write name content
+  | None -> Bos.OS.File.write filename content
```

Comment:
This can be rewritten `Bos.OS.File.write (Option.value outfile ~default:filename) content` FYI :)